overcast/runner/__init__.py

```diff
@@ -373,7 +415,7 @@ def wait():
 
     def shell_step_cmd(self, details, env_prefix=''):
         if details.get('type', None) == 'remote':
-             node = self.nodes[details['node']][1]
+              node = next(iter(self.nodes[details['node']].fip_addresses))
```

Comment:
This next(iter(some_set)) stuff is correct, but it looks awkward.

Two suggestions:
Either do something like

    for fip_address in self.fip_addresses: break

which will effectively do the exact same thing.

Or, don't make it a set. Just have a .fip_address attribute on Node objects that holds the last floating ip address assigned to a node. I can't think of a use case for multiple floating ip's on the same node anyway?